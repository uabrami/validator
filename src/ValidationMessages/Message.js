import React from "react";

const Message = ({ message }) =>{
  const isDecorated = message.isCompleted ? "line-through" : "";
 return (
  <li style={{textDecoration: isDecorated}}>
    {message.text}
  </li>
 );
};

export default Message;
