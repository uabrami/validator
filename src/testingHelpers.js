import { render } from "@testing-library/react";

export const checkBasicRendering = (component) => {
  it("Expect to render", () => {
    const { container } = render(component);
    expect(container.firstChild).toBeDefined();
  });
};

export const checkRequiredProps = (component) => {
  it("Expect to have all required propTypes", () => {
    const spy = jest.spyOn(global.console, "error");
    render(component);
    expect(spy).not.toHaveBeenCalled();
    spy.mockClear();
  });
};
