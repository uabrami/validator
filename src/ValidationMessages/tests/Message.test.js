import React from "react";
import { render } from "@testing-library/react";

import Message from "../Message";

import { checkBasicRendering, checkRequiredProps } from "../../testingHelpers";

const testProps = { message: "test" };

const Component = <Message {...testProps} />;

describe("<Message />", () => {
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });

  checkBasicRendering(Component);
  checkRequiredProps(Component);
});
