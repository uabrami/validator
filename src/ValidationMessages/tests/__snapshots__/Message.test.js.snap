// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Should render and match the snapshot 1`] = `
<li>
  <span />
</li>
`;
